optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/bi/CompositeBiJoiner.java

```diff
@@ -52,39 +55,29 @@
 
     @Override
     public Function<A, Object> getLeftMapping(int index) {
-        assertMappingIndex(index);
```

Comment:
Why can we get rid of these fail fasts? Because it throws an ArrayOutOfBoundsException?
If only OptaPlanner code can trigger that, then the ArrayOutOfBoundsException is certainly fine. If a user can do something wrong that easily triggers it, it might not be ideal. But I can't immediately think of something, so +1 to remove the assert.

Reply:
Retracting this comment. See below.